clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4321,6 +4328,91 @@ void handleFailure(Throwable throwable) {
+                                o.logEndOffset(),
+                                OptionalLong.of(o.lastFetchTimestamp()),
+                                OptionalLong.of(o.lastCaughtUpTimestamp())));
+                });
+                QuorumInfo info = new QuorumInfo(topicName, leaderId, voters, observers);
+                return info;
+            }
+
+            @Override
+            DescribeQuorumRequest.Builder createRequest(int timeoutMs) {
+                return new Builder(DescribeQuorumRequest.singletonRequest(
+                        new TopicPartition(METADATA_TOPIC_NAME, METADATA_TOPIC_PARTITION.partition())));
+            }
+
+            @Override
+            void handleResponse(AbstractResponse response) {
+                final DescribeQuorumResponse quorumResponse = (DescribeQuorumResponse) response;
+                try {
+                    if (quorumResponse.data().errorCode() != Errors.NONE.code()) {
+                        throw Errors.forCode(quorumResponse.data().errorCode()).exception();
+                    }
+                    if (quorumResponse.data().topics().size() > 1) {
+                        log.error("DescribeMetadataQuorum received {} topics when 1 was expected",
+                                quorumResponse.data().topics().size());
+                        throw new UnknownServerException();
+                    }
+                    DescribeQuorumResponseData.TopicData topic = quorumResponse.data().topics().get(0);
+                    if (!topic.topicName().equals(METADATA_TOPIC_NAME)) {
+                        log.error("DescribeMetadataQuorum received a topic with name {} when {} was expected",
+                                topic.topicName(), METADATA_TOPIC_NAME);
+                        throw new UnknownServerException();
+                    }
+                    if (topic.partitions().size() > 1) {
+                        log.error("DescribeMetadataQuorum received a topic {} with {} partitions when 1 was expected",
+                                topic.topicName(), topic.partitions().size());
+                        throw new UnknownServerException();
+                    }
+                    DescribeQuorumResponseData.PartitionData partition = topic.partitions().get(0);
+                    if (partition.partitionIndex() != METADATA_TOPIC_PARTITION.partition()) {
+                        log.error("DescribeMetadataQuorum received a single partition with index {} when {} was expected",
+                                partition.partitionIndex(), METADATA_TOPIC_PARTITION.partition());
+                        throw new UnknownServerException();
+                    }
+                    if (partition.errorCode() != Errors.NONE.code()) {
+                        throw Errors.forCode(partition.errorCode()).exception();
+                    }
+                    future.complete(createQuorumResult(quorumResponse));
```

Comment:
At this point, we have the `partition` that we are interested in. Couldn't we directly use it to populate `QuorumInfo` instead of looking it up again in `createQuorumResult`?